### DISCUSSION TITLE
Painfully slow performance on xrdp, but fast with gnome remote desktop

### DISCUSSION THREAD

**Discussion by t-anyu** [neutrinolabs/xrdp#2914] [Q&A]
Hey,

I have a Ubuntu 22.04 LTS server and a Windows RDP client that are joined with 1Gbps LAN connection.
I'm experiencing 0.5-1s lag. I have gone through all the optimization steps from internet:
- RFX codec
- different (smaller) resolutions
- tcp send buffer in xrdp.ini and kernel send buffer
- other xrdp.ini tricks
- 16bit, 24bit, 32bit depths
- router / switch QoS setups and etcetera
- Tried with different clients and macOS client

I also have compiled the latest xrdp, xorgxrdp with pulseaudio support, and the same problem persists.
The problem seems to be the xrdp as when using GNOME Remote Desktop RDP, the performance is silky smooth real-time responsiveness, which I want. But my server is intended to be headless, so no monitor.

Any idea how can I fix this lag? I have other setups where xrdp is performing like a charm.
Server is Ubuntu 22.04 LTS Desktop-minimal and using latest GNOME. Xrdp is using Xorg.

Thanks!

**Comment by t-anyu** [comment]
Hey,

I successfully launched xorg and gnome-session to a dummy display, then used freerdp-shadow-cli server as RDP server. The experience is order of a magnitude better performance. However I played around with display settings on xorg.conf, and it seems that certain modelines (refresh rates and other clocks) can degrade the performance to the same level as with xrdp.

I don't know if there is the issue.

**Comment by matt335672** [comment]
If you're using Windows 11, you won't get RFX. You'll have to wait until the GFX improvements we're working on get released. This is a significant project we're all looking at, so there will be more news soon.

Regarding send buffer sizes, see #2905 and #2910. This may not be relevant for you if your ping times are low anyway.

**Reply by t-anyu** [reply]
Hey, I've found another thing: I noticed that Gnome Remote Desktop used GPU. Then I compiled xorgxrdp with glamor support and forced it to use glamoregl and saw a huge performane DROP. I verified that when using without glamor support my intel_gpu_top showed 0% load on GPU, and when glamoregl loaded the GPU showed load.

Yes, I'm using Windows 11. Good to know. Thanks

**Reply by matt335672** [reply]
Stay tuned. We're working on getting GFX support out ASAP.

**Reply by Dariusz1989** [reply]
Is this now live? I'm having terrible perf :D in Ubuntu+Gnome+Win11+proxmox+pass_GPU :- )

**Reply by matt335672** [reply]
There's a beta of 0.10.x available on the releases page.

For Ubuntu, be aware that new versions of xrdp aren't backported. That's not our decision BTW.
